query_sheet.rb

```diff
@@ -186,5 +208,126 @@ def add_errors_to_message(
+  candidates.select do |candidate|
+    candidate[attribute] &&
+      candidate[attribute].length > 0
+  end.reduce({}) do |obj, candidate|
+    if obj[candidate[attribute]]
+      obj[candidate[attribute]] << candidate
+    else
+      obj[candidate[attribute]] = one_to_many ? [candidate] : candidate
+    end
+    obj
+  end
+end
+
+def find_diffs(imported_candidates)
+  imported_candidates_by_nationbuilder_id = candidates_by_attribute(
+    imported_candidates,
+    'nationbuilder_id',
+  )
+  sheets_data = assembly_district_sheets.reduce({}) do |obj, ad_sheet|
+    obj[ad_sheet['assembly_district']] = candidates_by_attribute(
+      read_sheet(ad_sheet['spreadsheet_id'], 'Candidate View'),
+      'Email',
+    )
+    obj
+  end
+  binding.pry
+end
+
+def column_to_letter(num)
+  char = (num % 26 + 65).chr
+  remainder = num/26
+  if num >= 26
+    column_to_letter(remainder - 1) + char
+  else
+    char
+  end
+end
+
+def get_ad_and_ed_from_cc_sunlight(address)
+  base_uri = 'https://ccsunlight.org/api/v1/address/'
+  uri = URI(base_uri + CGI.escape(address))
+  request = Net::HTTP::Get.new(path=uri)
+  response = Net::HTTP.start(uri.host, uri.port, use_ssl: true) do |http|
+    http.request(request)
+  end
+  ad, ed = JSON.parse(response.body).values_at('ad', 'ed')
+  {
+    ad: ad,
+    ed: ed,
+  }
+end
+
+def format_address(candidate)
+  unless candidate['primary_address1'] &&
+    candidate['primary_address1'].length > 0 &&
+    candidate['primary_city'] &&
+    candidate['primary_city'].length > 0 &&
+    candidate['primary_city'].match(/brooklyn|new york/i) &&
+    candidate['primary_state'] &&
+    candidate['primary_state'].length > 0 &&
+    candidate['primary_state'].match(/ny/i) &&
+    candidate['primary_zip'] &&
+    candidate['primary_zip'].length > 0 &&
+    candidate['primary_zip'].length == 5
+      return nil
+  end
+  "#{
+    titleize(candidate['primary_address1'])
+  }#{
+    if candidate['primary_address2'] && candidate['primary_address2'].length > 0
+      ' ' + candidate['primary_address2']
+    else
+      ''
+    end
+  }, #{
+    titleize(candidate['primary_city'])
+  }, #{
+    candidate['primary_state'].upcase
+  } #{
+    candidate['primary_zip']
+  }"
+end
+
+def titleize(str)
+  str
+    .to_s.split(/ |\_|\-/)
+    .map(&:capitalize)
+    .join(' ')
+end
+
+def map_address_ad_and_ed_to_candidates(candidates)
+  candidates.map do |candidate|
+    candidate['address'] = format_address(candidate)
+    if candidate['address']
+      puts candidate['address']
+      candidate['ad'], candidate['ed'] =
+        get_ad_and_ed_from_cc_sunlight(candidate['address'])
+          .values_at(:ad, :ed)
+    end
+    candidate
+  end
+end
+
+def nb_and_cc_ad_match?(candidiate)
+  candidate['state_lower_district'] == candidate['ad']
+end
+
+def ad_match?(candidiate, new_candidate)
+  candidate['ad'] == new_candidate['ad']
+end
+
+# spreadsheets_columns_valid?
+
+# find_diffs(
```

Comment:
Remove?

Reply:
yeah, might as well: it's kind of useless